realm/kotlin-extensions/src/main/kotlin/io/realm/kotlin/RealmObjectExtensions.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.realm.kotlin
+
+import io.realm.*
+import io.realm.RealmObject.freeze
+import io.realm.annotations.Beta
+import io.realm.internal.RealmObjectProxy
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.flow.flowOf
+
+/**
+ * Returns a [Flow] that monitors changes to this RealmObject. It will emit the current
+ * RealmObject when subscribed to. Object updates will continually be emitted as the RealmObject is
+ * updated - `onCompletion` will never be called.
+ *
+ * Items emitted from Realm flows are frozen - see [RealmObject.freeze]. This means that they are
+ * immutable and can be read from any thread.
+ *
+ * Realm flows always emit items from the thread holding the live RealmObject. This means that if
+ * you need to do further processing, it is recommended to collect the values on a computation
+ * dispatcher:
+ *
+ * ```
+ * object.toFlow()
+ *   .map { obj -> doExpensiveWork(obj) }
+ *   .flowOn(Dispatchers.IO)
+ *   .onEach { flowObject ->
+ *     // ...
+ *   }.launchIn(Dispatchers.Main)
+ * ```
+ *
+ * If your would like `toFlow()` to stop emitting items you can instruct the flow to only emit the
+ * first item by calling [kotlinx.coroutines.flow.first]:
+ * ```
+ * val foo = object.toFlow()
+ *   .flowOn(context)
+ *   .first()
+ * ```
+ *
+ * @return Kotlin [Flow] on which calls to `onEach` or `collect` can be made.
+ */
+@Beta
+fun <T : RealmModel> T.toFlow(): Flow<T> {
+    val obj = this
+    return if (obj is RealmObjectProxy) {
+        val proxy = obj as RealmObjectProxy
+        val realm = proxy.`realmGet$proxyState`().`realm$realm`
+
+        when (realm) {
+            is Realm -> flowFromRealm<T>(realm, obj)
+            is DynamicRealm -> {
+                val dynamicObject = obj as DynamicRealmObject
+                flowFromDynamicRealm(realm, dynamicObject) as Flow<T>
+            }
+            // FIXME: exception text
+            else -> throw UnsupportedOperationException("${realm.javaClass} does not support RxJava. See https://realm.io/docs/java/latest/#rxjava for more details.")
+        }
+    } else {
+        // TODO Is this true? Should we just return Observable.just(object) ?
```

Comment:
Good point. I can see that we disallow this for RxJava Observables as well. I cannot remember exactly why we did that.

This seems similar to frozen objects so probably fine to allow?